fvm/evm/handler/handler.go

```diff
@@ -224,6 +231,8 @@ func (h *ContractHandler) batchRun(rlpEncodedTxs [][]byte, coinbase types.Addres
 		if err != nil {
 			return nil, err
 		}
+
+		h.tracer.Collect(r.TxHash, h.blockID)
 	}
 
 	err = h.emitEvent(types.NewBlockEvent(bp))
```

Comment:
Still, the risk of tx failing because the emit event still exists, but I think for now is fine. maybe just add a comment about it. We might in the future do it in two phases commit and upload

Reply:
if the transaction still fails after and traces are uploaded it's not end of the world, because it's up to the consumer (EVM GW) to only get traces that succeeded (so if things error out, the block even wouldn't be emitted, which wouldn't download the trace).